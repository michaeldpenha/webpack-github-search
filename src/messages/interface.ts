export interface IMessages {
  noDataFound: string;
  noIssueFound: string;
  noRepositoryFound: string;
  noCommentFound: string;
  loadMore: string;
  searchLabel: string;
  repositoryLabel: string;
  stateLabel: string;
  homeLabel: string;
  lookUpDefaultText: string;
  errorLabel: string;
  issueFooterText: string;
  opened: string;
  closed: string;
  comments: string;
  issueLogLine: string;
  commentTitle: string;
  back: string;
  noContentAvailable: string;
  issueSearch: string;
  repoSearch: string;
}
