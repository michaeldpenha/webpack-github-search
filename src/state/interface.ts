import { ReactNode } from 'react';

export interface IHeaderInfo {
  title: string;
  goBack?: () => void;
  backText?: string;
}

export interface IAppContextProps {
  headerInfo?: IHeaderInfo;
}

export interface IAppStateContext {
  reducer: (state: IAppContextProps, action: Record<string, unknown>) => IAppContextProps | any;
  initialState: IAppContextProps;
  children: ReactNode;
}
