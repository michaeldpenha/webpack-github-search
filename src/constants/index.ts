export const PAGE_SIZE = 15;
export const BASE_FONT_SIZE = 10;

export const STATUS_OPTIONS = [
  {
    label: 'Open',
    value: 'open',
  },
  {
    label: 'Closed',
    value: 'closed',
  },
];

export const DEFAULT_ISSUE_SEARCH_QUERY = {
  type: 'issue',
  language: 'all',
  state: 'open',
};

export const DEFAULT_REPO_SEARCH_QUERY = {
  language: 'all',
};

export enum ROUTER {
  home = '/',
  issueDetail = '/:repoOwner/:repo/issues/:issueNumber',
  other = '*',
}

export enum StateActionTypes {
  headerInfo = 'HEADER_INFO',
}
