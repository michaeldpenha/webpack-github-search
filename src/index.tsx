import { ApolloProvider } from '@apollo/client';
import React from 'react';
import ReactDOM from 'react-dom';

import App from './app';
import { apolloClient } from './utils/apollo-client';

ReactDOM.render(
  <ApolloProvider client={apolloClient}>
    <App />
  </ApolloProvider>,
  document.getElementById('root'),
);
