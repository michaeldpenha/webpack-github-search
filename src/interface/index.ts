import { ReactChild, ReactNode, ReactElement } from 'react';

export type TuppleType = 'issue' | 'comment' | 'repo';
export type LoaderType = 'content' | 'list';
export type Typename = 'Repository' | 'Issue' | 'IssueComment';

export interface IData {
  id: string;
  __typename?: Typename | string;
  bodyHTML?: string;
  createdAt?: string;
  author?: IActor;
  owner?: IActor;
  nameWithOwner?: string;
  closed?: boolean;
  closedAt?: string;
  number?: number;
  repository?: IRepo;
  title?: string;
  body?: string;
}

export interface IRepo {
  id: string;
  name: string;
  owner: IActor;
  nameWithOwner: string;
}

export interface IActor {
  login: string;
  avatarUrl: string;
}

export interface ITuppleLoaderTypes {
  type: TuppleType;
  loaderType: LoaderType;
  displayNoDataText?: boolean;
  noDataFoundText?: string;
}

export interface ILists extends ITuppleLoaderTypes {
  data: IData[];
  loading: boolean;
  showLoadMore?: boolean;
  loadMore?: () => void;
  onItemClick?: (data: IData) => void;
}

export interface ITupple {
  data: IData;
  onTuppleClick?: (data: IData) => void;
  type?: TuppleType;
}

export interface ILayout {
  children: ReactChild | ReactNode | ReactElement;
}

export interface IDropDownData {
  label: string;
  value: string;
}

export interface IDropDown {
  data: IDropDownData[];
  defaultValue?: string;
  onSelectionChange?: (data: IDropDownData) => void;
}

export interface ISearch {
  defaultValue: string;
  autoComplete?: string;
  placeholder?: string;
  onInputChange?: (e: string) => void;
  onFocus?: (val: boolean) => void;
  onClick?: () => void;
}

export interface ILookUpData {
  id: string;
  nameWithOwner: string;
  name: string;
}

export interface ILookUp extends ITuppleLoaderTypes {
  data: IData[];
  defaultValue: string;
  loadMore: () => void;
  showMore: boolean;
  loading?: boolean;
  onClick?: (item?: IData) => void;
  onChange?: (val: string) => void;
  placeholder?: string;
}

export interface ILoader {
  display: boolean;
  type: LoaderType;
}
