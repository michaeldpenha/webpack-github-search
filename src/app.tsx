import React from 'react';
import { Router } from 'react-router';
import { createBrowserHistory } from 'history';
import { GlobalStyle } from './styles/global';
import Header from 'components/header';
import Layout from 'components/layout';
import Routes from 'components/router';
import { ThemeProvider } from 'styled-components';
import { theme } from './styles/theme';
import { StateProvider } from './state/context';
import { appReducer } from './state/reducer';
import { IAppContextProps } from './state/interface';

const history = createBrowserHistory({
  basename: '/',
});

const App: React.FC = () => {
  const appContext: IAppContextProps = {
    headerInfo: {
      title: '',
    },
  };

  return (
    <Router history={history}>
      <GlobalStyle />
      <ThemeProvider theme={theme}>
        <StateProvider initialState={appContext} reducer={appReducer}>
          <Header />
          <Layout>
            <Routes />
          </Layout>
        </StateProvider>
      </ThemeProvider>
    </Router>
  );
};

export default App;
