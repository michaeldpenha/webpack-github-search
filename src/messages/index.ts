import { IMessages } from './interface';

export const MESSAGES: IMessages = {
  noDataFound: 'No data found',
  noIssueFound: 'No issues found',
  noRepositoryFound: 'No repository found',
  noCommentFound: 'No comments found',
  loadMore: 'Load More',
  searchLabel: 'Search',
  repositoryLabel: 'Repository',
  stateLabel: 'State',
  homeLabel: 'Github Issues',
  lookUpDefaultText: 'Please type in to search',
  errorLabel: 'Errors',
  issueFooterText: 'Created by {0} {1} {2}',
  opened: 'Opened',
  closed: 'Closed',
  comments: 'Comments',
  issueLogLine: 'Created by {0} {1}',
  commentTitle: '{0} commented {1}',
  back: 'Back',
  noContentAvailable: 'No Content available',
  issueSearch: 'Search issue by name or body',
  repoSearch: 'Search Repo by name',
  issueCount: '{0} Issues found for {1} in {2} state',
  githubRepos: 'All Github repos',
};
