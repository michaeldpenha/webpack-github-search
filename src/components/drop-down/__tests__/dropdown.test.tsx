import React from 'react';
import { act } from 'react-dom/test-utils';
import { STATUS_OPTIONS } from 'src/constants';
import { IDropDown, IDropDownData } from 'src/interface';
import renderEl, { fireEvent } from 'src/testUtils/tests-render';
import DropDown from '..';

const setup = ({ data }: IDropDown) => renderEl(<DropDown data={data} defaultValue="Open" />);

describe('Drop Down component', () => {
  it('renders the dropdwon component', () => {
    const { getByTestId, queryByTestId } = setup({
      data: STATUS_OPTIONS,
    });
    const dropDownEl = getByTestId('drop-down-selected-value');

    expect(getByTestId('drop-down-container')).toBeVisible();
    expect(dropDownEl).toHaveTextContent('Open');

    fireEvent.click(dropDownEl);
    expect(queryByTestId('drop-down-list')).toBeInTheDocument();
    fireEvent.click(queryByTestId('drop-down-1'));

    expect(dropDownEl).toHaveTextContent('Closed');
    expect(queryByTestId('drop-down-list')).not.toBeInTheDocument();

    expect(dropDownEl).toMatchSnapshot();
  });
});
