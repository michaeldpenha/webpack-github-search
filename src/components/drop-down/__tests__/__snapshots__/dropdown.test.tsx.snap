// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drop Down component renders the dropdwon component 1`] = `
<div
  class="styles__DropDownHeader-sc-1c97jj7-1 dPLnHw"
  data-testid="drop-down-selected-value"
>
  Closed
</div>
`;
